golden-ratio.el

```diff
@@ -120,72 +120,39 @@ will not cause the window to be resized to the golden ratio."
       (when (window-resizable-p (selected-window) ncol t)
         (enlarge-window ncol t)))))
 
+(defvar golden-ratio-in-progress nil
+  "Avoid recursive adjustment.")
+
 ;;;###autoload
 (defun golden-ratio ()
   "Resizes current window to the golden-ratio's size specs."
   (interactive)
-  (unless (or (window-minibuffer-p)
-              (one-window-p)
-              (member (symbol-name major-mode)
-                      golden-ratio-exclude-modes)
-              (member (buffer-name)
-                      golden-ratio-exclude-buffer-names)
-              (and golden-ratio-inhibit-functions
-                   (loop for fun in golden-ratio-inhibit-functions
-                         thereis (funcall fun))))
-    (let ((dims (golden-ratio--dimensions))
-          (golden-p (if golden-ratio-mode 1 -1)))
-      ;; Always disable `golden-ratio-mode' to avoid
-      ;; infinite loop in `balance-windows'.
-      (golden-ratio-mode -1)
-      (balance-windows)
-      (golden-ratio--resize-window dims)
-      (when golden-ratio-recenter
-        (scroll-right) (recenter))
-      (golden-ratio-mode golden-p))))
-
-;; Should return nil
-(defadvice other-window
-    (after golden-ratio-resize-window)
-  (golden-ratio) nil)
-
-;; Should return the buffer
-(defadvice pop-to-buffer
-    (around golden-ratio-resize-window)
-  (prog1 ad-do-it (golden-ratio)))
-
-(defun golden-ratio--post-command-hook ()
-  (when (or (memq this-command golden-ratio-extra-commands)
-            (and (consp this-command) ; A lambda form.
-                 (loop for com in golden-ratio-extra-commands
-                       thereis (or (memq com this-command)
-                                   (memq (car-safe com) this-command)))))
-    ;; This is needed in emacs-25 to avoid this error from `recenter':
-    ;; `recenter'ing a window that does not display current-buffer.
-    ;; This doesn't happen in emacs-24.4 and previous versions.
-    (run-with-idle-timer 0.01 nil (lambda () (golden-ratio)))))
-
-(defun golden-ratio--mouse-leave-buffer-hook ()
-  (run-at-time 0.1 nil (lambda ()
-			 (golden-ratio))))
+  (when (and golden-ratio-mode
+             (not golden-ratio-in-progress))
+    (let ((golden-ratio-in-progress t))
+      (unless (or (window-minibuffer-p)
+                  (one-window-p)
+                  (member (symbol-name major-mode)
+                          golden-ratio-exclude-modes)
+                  (member (buffer-name)
+                          golden-ratio-exclude-buffer-names)
+                  (and golden-ratio-inhibit-functions
+                       (loop for fun in golden-ratio-inhibit-functions
+                          thereis (funcall fun))))
+        (balance-windows)
+        (golden-ratio--resize-window (golden-ratio--dimensions))
+        (when golden-ratio-recenter
+          (scroll-right) (recenter))))))
+
+(defadvice select-window (after golden-ratio-select-window activate)
+  (when golden-ratio-mode
```

Comment:
My 2 cents,

Should not happen, if the mode is disabled then the advice should be removed as well to keep a clear state at all time.

Consider introducing a variable `inhibit-golden-ratio` instead which can be used in a `let` form by the user and can also be used used as a flag to remove the need of `golden-ratio-in-progress` re-entrance variable.